### DISCUSSION TITLE
v2.0.0

### DISCUSSION THREAD

**Discussion by pelletier** [pelletier/go-toml#763] [Releases]
It is finally here! go-toml v2 is now stable. :tada:

Only just over one year after I started working on it. This version of go-toml has been rebuilt from the ground up. It respects the latest TOML specification. Countless bugs have been fixed. It behaves more like the standard library. And it is [_much_ faster](https://github.com/pelletier/go-toml/tree/v2#benchmarks). Special attention was given to keeping the public API to a minimum. Continuing the strict policy of backward compatibility, the next breaking change will be in another 5 years!

Thank you to everybody who contributed to this new version, as well as folks willing to try out the intermediate beta versions. Shout out to @moorereason for reporting many issues during the development process using differential fuzzing, as well as @bep for running a beta on the popular [Hugo](https://gohugo.io/) project!

There is still some work that could be done to improve the performance of the library (https://github.com/pelletier/go-toml/pull/758 and https://github.com/pelletier/go-toml/pull/669 for instance). But I am confident that the library is in a good-enough place at the moment for folks to reap the benefits of the rewrite. The major feature removal is the sort-of-AST that v1 provided. It allowed manipulating an arbitrary document and writing it back out. This was done to cut down the scope of the project, removing a feature that had unclear use-cases. The v2 parser creates some [intermediate AST](https://github.com/pelletier/go-toml/blob/v2/internal/ast/ast.go) so that if somebody is up to design a Document API the tools should be available to do so. Please reach out if you are interested!

The readme contains a [list of differences](https://github.com/pelletier/go-toml/tree/v2#migrating-from-v1) between v1 and v2 to be aware of when upgrading. Please report any issue you may have with the new version in the [bug tracker 🐞](https://github.com/pelletier/go-toml/issues), provide feedback, and ask questions in [Discussions 💬](https://github.com/pelletier/go-toml/discussions).
With this release, go-toml **v1 will not be receiving any updates**. This is required to keep maintenance of the project to a minimum. 

Hopefully, this change doesn't break your thing!

---

**Full commit log**: https://github.com/pelletier/go-toml/compare/c9a09d8695a83a55cd171d147b09fdf10270e503...v2

Thank you contributors: @pelletier @oschwald @moorereason @xwjdsh @kkHAIKE @RiyaJohn @jidicula @zostay @mmorel-35.

<hr /><em>This discussion was created from the release <a href='https://github.com/pelletier/go-toml/releases/tag/v2.0.0'>v2.0.0</a>.</em>

**Comment by moorereason** [comment]
Woohoo!  Great work, Thomas.  You've put a ton of work into this project, and I've been impressed with how great v2 turned out.  I honestly didn't think the parser would turn out as fast and efficient as it did, but major kudos to you for proving me wrong.  🏆
